### ISSUE TITLE
Out of memory error with scribble

### ISSUE THREAD

**Issue by zenspider** [racket/racket#1719]
Running `raco scribble index.rkt` on the following:

```racket
#lang scribble/manual
@(require racket/sandbox)
@(define evaluator (make-evaluator 'racket))
@(define typed/evaluator (make-evaluator 'typed/racket))
```

gives:

```
evaluator: terminated (out-of-memory)
  context...:
   /MyApplications/dev/lisp/Racket/share/pkgs/sandbox-lib/racket/sandbox.rkt:911:2: user-eval
...
```

@greghendershott figured out that upping the memory limit to 100 "fixes" it.

**Comment by greghendershott** [comment]
Context: This came up on IRC yesterday when @zenspider cloned [Fear of Macros](https://github.com/greghendershott/fear-of-macros) and tried to `make html-single`.

I hadn't updated FoM and tried to `make html-single` in quite awhile -- possibly two or more years! I'm pretty sure it used to work, last I tried. But when I tried, yesterday, I got the same error.

I avoid the error if I parameterize `sandbox-memory-limit` to something bigger than the documented default `30`.  I took a WAG and found `100` works:

```diff
modified   index.rkt
@@ -12,12 +12,14 @@
           (for-label racket/syntax))
 @(define evaluator
    (parameterize ([sandbox-output 'string]
-                  [sandbox-error-output 'string])
+                  [sandbox-error-output 'string]
+                  [sandbox-memory-limit 100])
      (make-evaluator 'racket)))
 
 @(define typed/evaluator
    (parameterize ([sandbox-output 'string]
-                  [sandbox-error-output 'string])
+                  [sandbox-error-output 'string]
+                  [sandbox-memory-limit 100])
      (make-evaluator 'typed/racket)))
```

So now "it works".

It's interesting (to me) that the memory needed has apparently increased over one or more versions of Racket. On the face of it that's not shocking (to me). Maybe nothing actionable here. Unless maybe if the increase indicates some other issue that needs attention?

Or maybe just `sandbox-memory-limit` should default higher now?

Or maybe the `make-evaluator` docs need to mention this?

I don't know enough to say for sure which if any of these.

TL;DR: Later versions of Racket need more memory to do something. Either this matters, or it doesn't.

**Comment by zenspider** [comment]
See also: https://github.com/racket/drracket/issues/123

**Comment by greghendershott** [comment]
By the way, I tried [building FoM using Travis CI against Racket versions from 5.3.5 through 6.8](https://travis-ci.org/greghendershott/fear-of-macros/builds/241736983). There _wasn't_ any simple pattern such as "the default 30 MB is inadequate only starting with vX.X".

---

Anyway, for Fear of Macros I think it is fine just to set `sandbox-memory-limit` to `#f`, and I'll push a commit to do that. (Although again, this repo is what I use to build part of my site, not intended as a library for others to use.)

**Comment by mflatt** [comment]
I agree that setting `sandbox-memory-limit` to `#f` is the right choice here. 

Avoiding memory-use increase is a constant battle, which I think explains why there isn't a simple pattern for which version requires 30 MB. In fact, as I look at http://build-plot.racket-lang.org/, I see a spike at the left of the green and blue areas that seems to have appeared on June 2... so, we'll have another skirmish today or tomorrow.

Unfortunately, my short-term solution to the problem is to make it at least 3 times as difficult by having three different Racket variants in development.

**Comment by mflatt** [comment]
I'm not sure there's anything to do here (although we are down to just _two_ variants of Racket). Should we keep this open, or close it?

**Comment by greghendershott** [comment]
Although I didn't open the issue per se and I'll defer to @zenspider I think it's fine to close.
